src/db/tables/subscribers.ts

```diff
@@ -429,6 +414,85 @@ async function getSubscribersWaitingForMonthlyEmail(
+      "2022-01-01T00:00:00.000Z",
+  );
+
+  if (!flag?.is_enabled) {
+    return [];
+  }
+
+  let query = knex("subscribers")
+    .select<SubscriberRow[]>("subscribers.*")
+    .leftJoin(
+      "subscriber_email_preferences",
+      "subscribers.id",
+      "subscriber_email_preferences.subscriber_id",
+    )
+    // Only send to users who haven't opted out of the monthly activity email...
+    // (Note that the `monthly_monitor_report` column is re-used for both the Plus
+    // user activity email, and the free user activity email.)
+    // It looks like Knex's `.where` type definition doesn't accept Promise-returning
+    // functions, even though the code does; hence the `eslint-disable`)
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    .where((builder) =>
+      builder
+        .whereNull("monthly_monitor_report_free")
+        .orWhere("monthly_monitor_report_free", true),
+    )
+    // ...who haven't received the email in the last 1 month...
+    // It looks like Knex's `.where` type definition doesn't accept Promise-returning
+    // functions, even though the code does; hence the `eslint-disable`)
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    .andWhere((builder) =>
+      builder
+        .whereNull("monthly_monitor_report_free_at")
+        .orWhereRaw(
+          "\"monthly_monitor_report_free_at\" < NOW() - INTERVAL '1 month'",
+        ),
+    )
+    // ...whose account is older than 1 month...
+    .andWhereRaw("subscribers.\"created_at\" < NOW() - INTERVAL '1 month'")
+    // ...but is no older than three years...
+    .andWhere("subscribers.created_at", ">=", accountCutOffDate)
+    // ...and who do not have a Plus subscription.
+    // Note: This will only match people of whom the Monitor database knows that
+    //       they have a Plus subscription. SubPlat is the source of truth, but
+    //       our database is updated via a webhook and whenever the user logs
+    //       in. Locally, you might want to set this via `/admin/dev/`.
+    // It looks like Knex's `.where` type definition doesn't accept Promise-returning
+    // functions, even though the code does; hence the `eslint-disable`)
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    .andWhere((builder) =>
+      builder
+        .whereRaw(
+          `NOT (subscribers.fxa_profile_json)::jsonb \\? 'subscriptions'`,
+        )
+        .orWhereRaw(
+          `NOT (subscribers.fxa_profile_json->'subscriptions')::jsonb \\? ?`,
+          MONITOR_PREMIUM_CAPABILITY,
+        ),
+    );
+
+  if (Array.isArray(flag.allow_list) && flag.allow_list.length > 0) {
+    // If the feature flag has an allowlist, only send to users on that list.
+    // The `.andWhereIn` alias doesn't exist:
+    // https://github.com/knex/knex/issues/1881#issuecomment-275433906
+    query = query.whereIn("subscribers.primary_email", flag.allow_list);
+  }
+  const rows = await query;
+
+  return rows;
```

Comment:
I kind of like this pattern vs. the usual `res[0]`, makes it a bit clearer what's being returned.